### ISSUE TITLE
Bug when using nimble 1.2.0 

### ISSUE THREAD

**Issue by ercrema** [ercrema/baorista#3]
Latest version of nimble (1.2.0) generates the following error message when executing the `expfit()` function:

```
library(baorista)
data(samplemat) 
x.mat <- createProbMat(pmat=samplemat,timeRange=c(5000,3001),resolution=100)
exponential.fit <- expfit(x.mat) #fitting using default MCMC/Prior settings
```

> Compiling nimble model...
> Error in FUN(X[[i]], ...) : 
>   checkAndPrepareDistributionInfo: random generation function `rAExp` is missing `returnType` or `returnType` does not match the type of the `x` argument to the corresponding density function.

The problem is most likely caused by inconsistencies between the random number generator function and the custom probability distribution.
Downgrading nimble to 1.1.0 fixes the problem (e.g. running `devtools::install_version('nimble',version='1.1.0')`) , but an update of `expfit()`, `logisticfit()` and `icarfit()` is required, or alternatively the issue is with nimble 1.2.0 itself. nimble developers have been contacted for help.

**Comment by ercrema** [comment]
Commit 7ed80a4d826e52123b8d46e12174e022d5a153ab on the development branch for 0.1.4 solves the issue, but generates a `Undefined global functions or variables` NOTE on cran check. A possible solution workaround might be to:

1. Move and declare or custom distribution outside the key functions
2. Supply as an argument for `parLapply()` all the custom distribution avoiding to declare them internally

**Comment by ercrema** [comment]
Fixed with [2bb595b](https://github.com/ercrema/baorista/commit/2bb595b48c62d0e15417facce508f7bb327cb0b4)
